lib/railway_ui_web/live/message_live/index.ex

```diff
@@ -0,0 +1,90 @@
+defmodule RailwayUiWeb.MessageLive.Index do
+  alias RailwayUiWeb.MessageLive.Index.State
+  alias RailwayUiWeb.Router.Helpers, as: Routes
+
+
+  defmacro __using__(opts) do
+    quote do
+      use Phoenix.LiveView
+      @message_type unquote(opts)[:message]
+      @view unquote(opts)[:view]
+
+      def render(assigns) do
+        Phoenix.View.render(@view, "index.html", assigns)
+      end
+
+      def mount(%{current_user_uuid: current_user_uuid}, socket) do
+        socket =
+          socket
+          |> assign(:messages, State.load_messages(@message_type))
```

Comment:
Messages have to be top-level or else we can't use temporary assigns. Temporary assigns means we won't store messages in memory.